<a href="https://colab.research.google.com/github/Danielfilho90/livro-receitas/blob/master/C%C3%B3pia_de_%5BE%5D_Aula_01_Tipos_de_Dados_e_Operadores_Aritm%C3%A9ticos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img src="https://www.cesar.school/wp-content/uploads/2019/09/marca_cesar_school.png" alt="drawing" width="200"/>

# Python - **Aula 01**

---

- Entender conceitos básicos sobre programação;
- Conhecer a linguagem Python;
- Criar seu primeiro programa em Python;

#Compreendendo as variáveis

Em termos técnicos, uma variável atribui um local de armazenamento a um valor que está vinculado a um nome ou identificador simbólico. O nome da variável é usado para fazer referência a esse valor armazenado em um programa de computador.

In [ ]:
my_int = 103204934813
age = 31

*   o nome da variável `my_int`
*   o operador de atribuição, também conhecido como sinal de igual `=`
*   o valor que está sendo vinculado ao nome da variável `103204934813`

## Regras para criação de nomes de variáveis:

- Não pode começar com número;
- Não pode haver espaço (use `_`);
- Não pode haver caracteres especiais: `" ' , < > | / \ ( ) @ ! # $ % ^ & * ~ - +`

##Reatribuindo Variáveis

##Atribuição múltipla

Com Python, você pode atribuir um único valor a várias variáveis ​​ao mesmo tempo. Isso permite inicializar várias variáveis ​​de uma vez, que você mesmo pode reatribuir posteriormente no programa ou por meio da entrada do usuário.

In [ ]:
x = y = z = 0
print(x)
print(y)
print(z)

0
0
0



#Criação e impressão de strings

Strings existem entre aspas simples `'` ou duplas `"` em Python, portanto, para criar uma string, coloque uma sequência de caracteres em um ou outro.

In [ ]:
'Esta é uma string entre aspas simples.'
"Esta é uma string entre aspas duplas."

Podemos imprimir strings simplesmente chamando a função `print()`

In [ ]:
# Print em Python


##Sintaxe do comentário
Os comentários em Python começam com uma marca de hash `#` e um caractere de espaço em branco e continuam até o final da linha.

Geralmente, os comentários serão semelhantes a este:

In [ ]:
# comentário de linha
'''comentário de bloco'''

##Comentários Inline
Os comentários Iinline ocorrem na mesma linha de uma instrução, seguindo o próprio código.

Geralmente, os comentários inline são assim:

In [ ]:
x = 8

# Tipos de dados

## Numéricos
- `int` - Inteiros
- `float` - Pontos Flutuantes

## Texto
- `str` - String

## Booleano
- `bool` - `True` ou `False`

##Listas
- `list` - [ ]

##Tuplas
- `tuples` - ( )

##Dicionários
- `dictionary` - { }

> Função `type()`

In [ ]:
print(type(10))
print(type(10.0))
print(type('oi'))
print(type(True))

<class 'int'>
<class 'float'>
<class 'str'>
<class 'bool'>


# ENTRADA DE DADOS
Entrada de Dados refere-se a entrada de dados do mundo externo para o meio digital.

Sintaxe da função input()

```variável = input(mensagem opcional)```


In [ ]:
name = input('Insira seu nome ')
print('O meu nome é', name)
input()

Insira seu nome Daniel
O meu nome é Daniel
Daniel


'Daniel'

#Funções para conversão de tipo

Algumas vezes é necessário converter valores de um tipo para outro. As funções int, float e str tentarão converter seus argumentos para os tipos int, float e str, respectivamente. Nós chamamos elas de funções para conversão de tipos.

##Convertendo os tipos de número

In [ ]:
# Convertendo os números inteiros em floats
a = 10
print(type(a))
a = float('10.5')
print(type(a))
print(a)

# Convertendo floats em inteiros
b = int(15.0)
print(b)


# Números convertidos através de divisão
a = 5 / 3
print(a)

<class 'int'>
<class 'float'>
10.5
15
1.6666666666666667


##Convertendo com as strings

In [ ]:
# Convertendo números em strings
txt = str(50)
txt = str(txt)
print(type(txt))

# Convertendo strings em números
year = int(input())

print(2021 - year)

# --------------------------


<class 'str'>
31
1990


# Operações Aritméticas

`x + y`	Soma de X e Y

`x - y`	Subtração de X e Y

`-x`	Sinal alterado de X

`+x`	Identidade de X

`x * y`	Multiplicação de X e Y

`x / y`	Divisão de X e Y (aqui tudo será dividido para float)

`x // y`	Divisão de X e Y com resultado em inteiro

`x % y`	Resto da divisão de X / Y

`x ** y`	X elevado a Y

In [ ]:
print(5 / 2)

2


In [ ]:
print(5 % 2)

1


In [ ]:
print(2 ** 4)

16


In [ ]:
n = 10
print(-n)

-10


In [ ]:
n = -10
print(+n)

-10


#Operações Aritméticas Unárias

Uma expressão matemática unária consiste em apenas um componente ou elemento e, em Python, os sinais de mais e menos podem ser usados ​​como um único elemento emparelhado com um valor para retornar a identidade do valor (`+`) ou alterar o sinal do valor (`-`).

A ordem da operação aritmética importa!

In [ ]:
total1 = 2 + 10 * 10 - 5
print(total1)

97


In [ ]:
total2 = (2 + 10) * (10 - 5)
print(total2)

60


In [ ]:
total3 = 20 + 5 * 2;
print(total3)

30


In [ ]:
total4 = (20 + 5) * 2;
print(total4)

50


In [ ]:
total5 = 20 / 2 * 5;
print(total5)

50.0


In [ ]:
total6 = 10 + 10 * 2 + 20 / 2;
print(total6)

40.0


##Operadores de atribuição


* `+=` - - adiciona e atribui o valor
* `-=` - - subtrai e atribui o valor
* `*=` - - multiplica e atribui o valor
* `/=` - - divide e atribui o valor
* `//=` - - divide e depois atribuir o valor em inteiro
* `**=` - - aumentar a potência e então atribuir o valor
* `%=` - - retorna o resto da divisão e depois atribui o valor



In [ ]:
y = 0
y += 1
y -= 1
y *= 2
y /= 3
y //= 5
y **= 2
y %= 3

0.0


# Exercícios

1. Faça um Programa que mostre a mensagem "Alo mundo" na tela

In [ ]:
print('Ola mundo')

Ola mundo



2. Faça um Programa que peça um número e então mostre a mensagem O número informado foi [número].

In [ ]:
number = input('Informe um numero ')
print('O numero informado foi', number)

Informe um numero 12
O numero informado foi 12



3. Faça um Programa que peça dois números e imprima a soma.

In [ ]:
number1 = int(input('Informe um numero '))
number2 = int(input('Informe um numero '))
print('a soma dos numeros é', number1 + number2)

Informe um numero 10
Informe um numero 20
a soma dos numeros é 30



4. Faça um Programa que peça as 4 notas bimestrais e mostre a média.

In [ ]:
nota1 = float(input('Informe sua primeira nota '))
nota2 = float(input('Informa sua segunda nota '))
nota3 = float(input('Informa sua terceira nota '))
nota4 = float(input('Informa sua quarta nota '))
soma = nota1 + nota2 + nota3 + nota4
media = soma / 4
print('Sua media foi ', media)

Informe sua primeira nota 10
Informa sua segunda nota 9
Informa sua terceira nota 8
Informa sua quarta nota 7
Sua media foi  8.5



5. Faça um Programa que converta metros para centímetros.

In [ ]:
altura = float(input('Digite sua altura '))
print('sua altura é ', altura * 100, 'cm')

Digite sua altura 1.75
sua altura é  175.0 cm



6. Faça um Programa que peça o raio de um círculo, calcule e mostre sua área.

In [ ]:
raio = float(input('Digite o raio da circunferencia '))
print('A area total da circunferencia ', raio * 2, 'm')

Digite o raio da circunferencia 2
A area total da circunferencia  4.0 m



7. Faça um Programa que calcule a área de um quadrado, em seguida mostre o dobro desta área para o usuário.

In [1]:
base = float(input('Escreva o tamanho da base '))
lado1 = float(input('Escreva o tamanho do lado 1 '))
lado2 = float(input('Escreva o tamanho lado 2 '))
superior = float(input('Escreva o tamanho do superior '))
soma = base + lado1 + lado2 + superior
print('o total do quadrado é ', soma)
print('\n o dobro da área é ', soma * 2)


Escreva o tamanho da base 10
Escreva o tamanho do lado 1 8
Escreva o tamanho lado 2 8
Escreva o tamanho do superior 10
o total do quadrado é  36.0

 o dobro da área é  72.0



8. Faça um Programa que pergunte quanto você ganha por hora e o número de horas trabalhadas no mês. Calcule e mostre o total do seu salário no referido mês.

In [ ]:
salario = float(input('Insira quanto você recebe por hora '))
horas = float(input('Quantas horas você trabalha por mês '))
print('Seu salário mensal é :','r$', salario * horas, 'reais')

Insira quanto você recebe por hora 12
Quantas horas você trabalha por mês 220
Seu salário mensal é : r$ 2640.0 reais



9. Faça um Programa que peça a temperatura em graus Fahrenheit, transforme e mostre a temperatura em graus Celsius.
> C = 5 * ((F-32) / 9).


10. Faça um Programa que peça a temperatura em graus Celsius, transforme e mostre em graus Fahrenheit.


11. Faça um Programa que peça 2 números inteiros e um número real. Calcule e mostre:
    - o produto do dobro do primeiro com metade do segundo.
    - soma do triplo do primeiro com o terceiro.
    - terceiro elevado ao cubo.


12. Tendo como dados de entrada a altura de uma pessoa, construa um algoritmo que calcule seu peso ideal, usando a seguinte fórmula: (72.7*altura) - 58


13. Tendo como dado de entrada a altura (h) de uma pessoa, construa um algoritmo que calcule seu peso ideal, utilizando as seguintes fórmulas:
    - Para homens: (72.7*h) - 58
    - Para mulheres: (62.1*h) - 44.7


14. João Papo-de-Pescador, homem de bem, comprou um microcomputador para controlar o rendimento diário de seu trabalho. Toda vez que ele traz um peso de peixes maior que o estabelecido pelo regulamento de pesca do estado de São Paulo (50 quilos) deve pagar uma multa de R\$ 4,00 por quilo excedente. João precisa que você faça um programa que leia a variável peso (peso de peixes) e calcule o excesso. Gravar na variável excesso a quantidade de quilos além do limite e na variável multa o valor da multa que João deverá pagar. Imprima os dados do programa com as mensagens adequadas.


15. Faça um Programa que pergunte quanto você ganha por hora e o número de horas trabalhadas no mês. Calcule e mostre o total do seu salário no referido mês, sabendo-se que são descontados 11% para o Imposto de Renda, 8% para o INSS e 5% para o sindicato, faça um programa que nos dê:
    - salário bruto.
    - quanto pagou ao INSS.
    - quanto pagou ao sindicato.
    - o salário líquido.
    - calcule os descontos e o salário líquido, conforme a tabela abaixo:

```
            + Salário Bruto : R$
            - IR (11%) : R$
            - INSS (8%) : R$
            - Sindicato ( 5%) : R$
            = Salário Liquido : R$
```
> Obs.: Salário Bruto - Descontos = Salário Líquido.



16. Faça um programa para uma loja de tintas. O programa deverá pedir o tamanho em metros quadrados da área a ser pintada. Considere que a cobertura da tinta é de 1 litro para cada 3 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam R\$ 80,00. Informe ao usuário a quantidades de latas de tinta a serem compradas e o preço total. Obs.: somente são vendidos um número inteiro de latas.


17. Faça um Programa para uma loja de tintas. O programa deverá pedir o tamanho em metros quadrados da área a ser pintada. Considere que a cobertura da tinta é de 1 litro para cada 6 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam R\$ 80,00 ou em galões de 3,6 litros, que custam R\$ 25,00.
  - Informe ao usuário as quantidades de tinta a serem compradas e os respectivos preços em 3 situações:
  - comprar apenas latas de 18 litros;
  - comprar apenas galões de 3,6 litros;
  - misturar latas e galões, de forma que o desperdício de tinta seja menor. Acrescente 10% de folga e sempre arredonde os valores para cima, isto é, considere latas cheias.


18. Faça um programa que peça o tamanho de um arquivo para download (em MB) e a velocidade de um link de Internet (em Mbps), calcule e informe o tempo aproximado de download do arquivo usando este link (em minutos).